app-doc/heirloom-doctools/heirloom-doctools-160308.ebuild

```diff
@@ -0,0 +1,90 @@
+EAPI=6
+
+inherit flag-o-matic toolchain-funcs readme.gentoo-r1
+
+DESCRIPTION="Classic Unix documentation tools ported from OpenSolaris"
+HOMEPAGE="https://n-t-roff.github.io/heirloom/doctools.html"
+SRC_URI="https://github.com/n-t-roff/heirloom-doctools/releases/download/${PV}/${P}.tar.bz2"
+
+LICENSE="CDDL"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="cxx"
+
+RDEPEND="!sys-apps/groff"
+DEPEND="sys-devel/flex
+	virtual/yacc"
+
+src_prepare() {
+	# mpm uses C++, we'll build it explicitly if we really want to
+	sed -i -e 's:mpm:$(MPM):' makefile || die
+
+	# Fixing compilation issues
+	# specific for versions =<160308, fixed in -git
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/eqn.d/Makefile.mk || die
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/eqn.d/Makefile.mk || die
+
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/neqn.d/Makefile.mk || die
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/neqn.d/Makefile.mk || die
+
+	eapply_user
+}
+
+src_configure() {
+	append-cppflags -D_GNU_SOURCE
+
+	sed \
+		-e "s:@CFLAGS@:${CFLAGS}:" \
+		-e "s:@CXXFLAGS@:${CXXFLAGS}:" \
+		-e "s:@CPPFLAGS@:${CPPFLAGS}:" \
+		-e "s:@LDFLAGS@:${LDFLAGS}:" \
+		-e "s:@CC@:$(tc-getCC):" \
+		-e "s:@CXX@:$(tc-getCXX):" \
+		-e "s:@RANLIB@:$(tc-getRANLIB):" \
+		-e "s:@libdir@:$(get_libdir):" \
+		"${FILESDIR}"/default.config \
+		> "${S}"/mk.config || die
```

Comment:
for most of these variables, its better to remove any assignments in the makefile and have make load them from the environment. If that's still to hardcore, use `?=` GNUism assignments in make, which will also first check in environment before taking defaults.